File: scripts/examples/auto_heal.py
# Автоматическое лечение при низком HP (Python)

# Настройки
HEAL_THRESHOLD = 30  # Процент HP для лечения
MANA_REQUIRED = 50   # Минимум маны для каста
HEAL_SPELL = "cast 'cure serious'"

def on_load(api):
    log("🏥 Автохил загружен (Python)")
    echo("═══════════════════════════════════════")
    echo("  Автохил активен! (Python/Jython)")
    echo("  Порог лечения: %d%%" % HEAL_THRESHOLD)
    echo("  Минимум маны: %d" % MANA_REQUIRED)
    echo("═══════════════════════════════════════")

def on_unload():
    log("🏥 Автохил выгружен")

def on_msdp(data):
    hp = api.getMsdpValue("HEALTH")
    max_hp = api.getMsdpValue("HEALTH_MAX")
    mana = api.getMsdpValue("MANA")

    if hp and max_hp and mana:
        try:
            hp_value = float(hp)
            max_hp_value = float(max_hp)
            mana_value = float(mana)

            hp_percent = (hp_value / max_hp_value) * 100

            if hp_percent < HEAL_THRESHOLD and mana_value > MANA_REQUIRED:
                echo("🏥 Автохил: HP %.0f%% - лечимся!" % hp_percent)
                send(HEAL_SPELL)
        except ValueError:
            pass

def on_line(line):
    # Реагируем на критическое состояние
    line_str = str(line)
    if "Вы истекаете кровью" in line_str or "Вы при смерти" in line_str:
        echo("⚠️  КРИТИЧЕСКОЕ СОСТОЯНИЕ!")
        send("flee")

File: scripts/examples/auto_heal.lua
-- Автоматическое лечение при низком HP (Lua)

-- Настройки
local HEAL_THRESHOLD = 30  -- Процент HP для лечения
local MANA_REQUIRED = 50   -- Минимум маны для каста
local HEAL_SPELL = "cast 'cure serious'"

function on_load(api)
    mud_log("🏥 Автохил загружен (Lua)")
    echo("═══════════════════════════════════════")
    echo("  Автохил активен! (Lua/LuaJ)")
    echo("  Порог лечения: " .. HEAL_THRESHOLD .. "%")
    echo("  Минимум маны: " .. MANA_REQUIRED)
    echo("═══════════════════════════════════════")
end

function on_unload()
    mud_log("🏥 Автохил выгружен")
end

function on_msdp(data)
    local hp = api:getMsdpValue("HEALTH")
    local max_hp = api:getMsdpValue("HEALTH_MAX")
    local mana = api:getMsdpValue("MANA")

    if hp and max_hp and mana then
        local hp_value = tonumber(hp)
        local max_hp_value = tonumber(max_hp)
        local mana_value = tonumber(mana)

        if hp_value and max_hp_value and mana_value then
            local hp_percent = (hp_value / max_hp_value) * 100

            if hp_percent < HEAL_THRESHOLD and mana_value > MANA_REQUIRED then
                echo(string.format("🏥 Автохил: HP %.0f%% - лечимся!", hp_percent))
                send(HEAL_SPELL)
            end
        end
    end
end

function on_line(line)
    -- Реагируем на критическое состояние
    local line_str = tostring(line)
    if string.find(line_str, "Вы истекаете кровью") or string.find(line_str, "Вы при смерти") then
        echo("⚠️  КРИТИЧЕСКОЕ СОСТОЯНИЕ!")
        send("flee")
    end
end
